index.html

```diff
@@ -899,57 +574,510 @@ <h3><dfn>RTCQuicTransportState</dfn> Enum</h3>
       </div>
     </section>
   </section>
+  <section id="quic-transport*">
+    <h2><dfn>QuicTransport</dfn> Interface</h2>
+    <p>The <code>QuicTransport</code> is a subclass of
+    <code>QuicTransportBase</code> focused on client/server use cases.</p>
+    <section id="quictransport-interface-definition*">
+      <h3>Interface Definition</h3>
+      <div>
+        <pre class="idl">
+        [ Constructor (DOMString url), Exposed=Window]
+interface QuicTransport : QuicTransportBase {
+};</pre>
+        <section>
+          <h2>Constructors</h2>
+          When the <code><a>QuicTransport</a></code> constructor is invoked,
+          the user agent <em class="rfc2119" title="MUST">MUST</em> run the
+          following steps:
+          <ol>
+            <li>Let <var>url</var> be the first argument.</li>
+            <li>Let <var>parsedUrl</var> be the result of appyling the 
+              <a href="https://url.spec.whatwg.org/#concept-url-parser">URL parser</a> to <var>url<var>.
+            </li>
+            <li>If <var>parsedUrl</var> is a failure, then throw a "SyntaxError" DOMException.</li>
+            <li>If <var>parsedUrl</var>'s scheme is not "quic", then throw a "SyntaxError" DOMException.</li>
+            <li>If <var>parsedUrl</var>'s username, password, path, query, or fragment is non-null,
+              then throw a "SyntaxError" DOMException.</li>
+            <li>
+              Let <var>quictransport</var> be a newly constructed
+              <code><a>QuicTransport</a></code> object.
+            </li>
+            <li>Let <var>quictransport</var> have a <dfn>[[\QuicTransportWritableStreams]]</dfn>
+            internal slot representing a sequence of <code><a>QuicWritableStream</a></code>
+            objects, initialized to empty.
+            </li>
+            <li>Let <var>quictransport</var> have a <dfn>[[\QuicTransportReadableStreams]]</dfn>
+            internal slot representing a sequence of <code><a>QuicReadableStream</a></code>
+            objects, initialized to empty.
+            </li>
+            <li>
+              Let <var>quictransport</var> have a <dfn>[[\QuicTransportState]]</dfn>
+              internal slot, initialized to <code>"connecting"</code>.
+            </li>
+            <li>Run these steps in parallel:
+              <ol>
+                <li>Establish a QUIC connection to the address identified by the
+                <var>parsedUrl</var>'s host and port.  The user agent MUST
+                include a QUIC transport parameter with ID web_origin(0x330) 
+                <!-- TODO: Register the transport parameter with IANA -->
+                and value equal to the value of an  <a
+                href="https://fetch.spec.whatwg.org/#origin-header">origin
+                header used for HTTP fetching</a></li>

```

Comment:
This should be defined more formally. Basically you need to obtain an origin somehow (probably from the relevant global object) and then serialize it. I don't think you need to reference Fetch for this, but you will need to reference HTML.

Reply:
Ah, thanks for pointing me in the right direction.  Did I get it right this time?

Reply:
No, you want to use similar prose as is used to define self.origin, not define it in terms of self.origin. Relevant global object is also relative to some other object, which I'm guessing should be the object being constructed here.

Reply:
Do you mean I should the phrase "the global object's origin"?  Or do I need to say something like "the origin of the global object relative to the QuicTransport being constructed"?

Reply:
OK, I went with that.  But the examples I saw in https://html.spec.whatwg.org/multipage/webappapis.html all said "relevant settings object" instead of "relevant global object".  I went with "relevant settings object".  Is that OK?  Or should I use  "relevant global object"?